### ISSUE TITLE
Task-scoped keys

### ISSUE THREAD

**Issue by eed3si9n** [sbt/sbt#202]
## steps

``` scala
> set test in Compile in compile := {}
> test
[error] Not a valid command: test (similar: set, last, exit)
[error] No such setting/task: {file:/Users/x/x/}default-b65acd/test
[error] test
[error]     ^
```
## problems
1. `test:test` is masked by `compile:test(for compile)`.
2. task-scoped keys cannot be discovered by tab completion.
## expectations
1. `test:test` runs.
2. task-scoped keys are denoted `project/config:task::key`.
3. `compile:[tab]` completes to `compile::test`.
## note

I can work around problem 1 as follows in 0.10, but not in 0.11:

``` scala
lazy val baseAssemblySettings: Seq[sbt.Project.Setting[_]] = Seq(
  test <<= test or (test in Test).identity,
  test in compile <<= (test in Test).identity,
)
```

**Comment by harrah** [comment]
Implemented in the parsing_changes branch:

https://github.com/harrah/xsbt/tree/parsing_changes

**Comment by harrah** [comment]
Included in the 0.12 branch.
